### PR TITLE
Fix pointer assignation in pod list + add chunking test

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

Fix an issue where we would send multiple times the same random pod because the pointer value was fixed at each iteration

Add a test to the chunking logic as a bonus while investigating the issue

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/util/kubernetes/kubelet/kubelet_orchestrator.go [hunk 1]
```diff
@@ -57,10 +57,11 @@ func (ku *KubeUtil) GetRawLocalPodList() ([]*v1.Pod, error) {
 	if !ok {
 		return nil, fmt.Errorf("pod list type assertion failed on %v", podListData)
 	}
+
 	// transform []v1.Pod in []*v1.Pod
-	pods := make([]*v1.Pod, 0, len(podList.Items))
-	for _, p := range podList.Items {
-		pods = append(pods, &p)
+	pods := make([]*v1.Pod, len(podList.Items))
```
Comment (shang-wang) [inline pkg/util/kubernetes/kubelet/kubelet_orchestrator.go:62]: Awesome find!

**Review by CharlyF — APPROVED** [review]

**Review by xlucas — APPROVED** [review]

File: pkg/process/util/orchestrator/pod_test.go [hunk 2]
```diff
@@ -496,3 +496,62 @@ func TestGetConditionMessage(t *testing.T) {
 		})
 	}
 }
+
+func TestChunkPods(t *testing.T) {
```
Comment (nammn) [inline pkg/process/util/orchestrator/pod_test.go:500]: great adding a test!

**Review by nammn — APPROVED** [review]
LGTM! 

IMO it might be worth to add the change to this unit test as well
https://github.com/DataDog/datadog-agent/blob/fd777ce4a3499b44b16e2f3687867f3bba42ffb2/pkg/util/kubernetes/kubelet/kubelet_orchestrator.go#L61

which just checks the length, but we may want to check that the pointers are different as well for that test. Helps with future regression (?).

I just run this on my local machine to verify. Your PR does fix that, the pointers are different each time. Checking with the version before the pods all had the same reference.

File: pkg/util/kubernetes/kubelet/kubelet_orchestrator_test.go [hunk 3]
```diff
@@ -64,6 +64,21 @@ func (suite *KubeletOrchestratorTestSuite) TestGetRawLocalPodList() {
 	pods, err := kubeutil.GetRawLocalPodList()
 	require.Nil(suite.T(), err)
 	require.Len(suite.T(), pods, 7)
+
```
Comment (nammn) [inline pkg/util/kubernetes/kubelet/kubelet_orchestrator_test.go:67]: 👍

**Review by nammn — APPROVED** [review]

**Pull request merged**